docs/HowToUsePeripheral.md

```diff
@@ -242,7 +242,58 @@ static void spi_select(struct spi_dev_s *dev, enum spi_dev_e devid, bool selecte
 
 }
 ```
+## I2C 
+
+Each board, shall;
+ 1. Expose an instanse of _struct i2c_dev_s_
+ 2. Implement the supported operations on i2c _struct i2c_op_s_
+
+```
+struct i2c_dev_s {
+	const struct i2c_ops_s *ops;	/* I2C vtable */
+	FAR void *priv;			/* Used by the arch-specific logic */
+};
+```
+
+```
+struct i2c_ops_s {
+	uint32_t (*setfrequency)(FAR struct i2c_dev_s *dev, uint32_t frequency);
+	int (*setaddress)(FAR struct i2c_dev_s *dev, int addr, int nbits);
+	int (*write)(FAR struct i2c_dev_s *dev, const uint8_t *buffer, int buflen);
+	int (*read)(FAR struct i2c_dev_s *dev, uint8_t *buffer, int buflen);
+#ifdef CONFIG_I2C_WRITEREAD
+	int (*writeread)(FAR struct i2c_dev_s *inst, const uint8_t *wbuffer, int wbuflen, uint8_t *rbuffer, int rbuflen);
+#endif
+
+#ifdef CONFIG_I2C_TRANSFER
+	int (*transfer)(FAR struct i2c_dev_s *dev, FAR struct i2c_msg_s *msgs, int count);
+#endif
+#ifdef CONFIG_I2C_SLAVE
+	int (*setownaddress)(FAR struct i2c_dev_s *dev, int addr, int nbits);
+
+	int (*registercallback)(FAR struct i2c_dev_s *dev, int (*callback)(void));
+#endif
+};
+
+```
+
+Above declarations are present in **[i2c.h](../os/include/tinyara/i2c.h)**
+
+In addition to above, each board level logic shall implement the 
+other functions listed in **[i2c.h](../os/include/tinyara/i2c.h)**
+
+```
+1. EXTERN FAR struct i2c_dev_s *up_i2cinitialize(int port);
+2. EXTERN int up_i2cuninitialize(FAR struct i2c_dev_s *dev);
+3. EXTERN int up_i2creset(FAR struct i2c_dev_s *dev);
+```
+
+### I2C Initialization
+I2C can be initialzed by invoking up_i2cinitialize 
+either as part of _up_intitialize()_ or as part of _board_initialize()_.
```

Comment:
1. Shall we provide a link to s5j  i2c driver (arch/arm/src/s5j/s5j_i2c.c) as a reference?
2. Low level I2C driver usually bound to higher level device driver to provide a device node to application layer. Hence shall we add details about "i2c_uioregister()" which does a register driver?

Reply:
Yes, worth mention about uio, will update this after updating the i2c code. defer this review for more than 2 weeks.